packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs

```diff
@@ -46,3 +48,43 @@ where
             .change_context(QueryError)
     }
 }
+
+#[async_trait]
+impl<C: AsClient, T> crud::Read<'_, (), T> for PostgresStore<C>
+where
+    for<'de> T: OntologyDatabaseType + Deserialize<'de> + Send,
+{
+    type Output = Vec<T>;
+
+    async fn get(&self, _: ()) -> Result<Self::Output, QueryError> {
+        // SAFETY: We insert a table name here, but `T::table()` is only accessible from within this
+        //   module.
+        let row_stream = self
+            .as_client()
+            .query_raw(
```

Comment:
As far as I understand using `query_raw` means exposing a stream of rows. Is that something we want to be doing for all of the 'aggregating' queries that return multiple rows back to the user? 

Since we're just collecting to a vec at the end, is there much benefit to this approach over simply fetching all rows out?

Reply:
I used `query_raw` here as I use the `Stream` later to get the content of the first column and deserialize it into `T`. Using `query` would call the same function, and collect it into a `Vec`. That Vec would need to be converted into an iterator again to read the column, so using `query_raw` avoids one `Vec` allocation.